src/Compilers/Core/CodeAnalysisTest/Diagnostics/SystemLanguageAnalyzer.cs

```diff
@@ -128,4 +132,118 @@ void Report(CompilationAnalysisContext context, IFieldSymbol field, DiagnosticDe
             context.ReportDiagnostic(Diagnostic.Create(descriptor, field.Locations.FirstOrDefault()));
         }
     }
+
+    /// <summary>Analyzer used to identify local variables that could be declared Const.</summary>
+    public class LocalCouldBeConstAnalyzer : DiagnosticAnalyzer
```

Comment:
There are quite a few more System C# analyzers to write. These are test-only analyzers for now, but we will probably promote them to real analyzers (and subject them to localization etc.) in the not-too-distant future.

I'll move at least the to-become-real ones to separate files.